src/CachePlugin.php

```diff
@@ -67,12 +68,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
+
+            return new FulfilledPromise($response);
         }
 
         return $next($request)->then(function (ResponseInterface $response) use ($cacheItem) {
             if ($this->isCacheable($response)) {
-                $cacheItem->set($response)
+                $cacheItem->set(['response' => $response, 'body' => $response->getBody()->__toString()])
```

Comment:
We need to have some sort of buffered stream, as some implementations provide a unbuffered stream, so reading a stream once will certainly make it not available afterwards.

Also even if the stream is buffered by the implementation, it must be rewind here.

Concretly we must add something like, if stream is seekable just rewind this, if not buffer it into a new stream.

We may face memory problem if big file are downloaded, but don't know if we can avoid that